AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -508,4 +508,17 @@ - (NSArray *)nodesAtIndexPaths:(NSArray *)indexPaths
   return ASFindElementsInMultidimensionalArrayAtIndexPaths(_nodes, [indexPaths sortedArrayUsingSelector:@selector(compare:)]);
 }
 
+#pragma mark - Dealloc
+
+- (void)dealloc {
+  ASDisplayNodeAssertMainThread();
+  [_nodes enumerateObjectsUsingBlock:^(NSMutableArray *section, NSUInteger sectionIndex, BOOL *stop) {
+    [section enumerateObjectsUsingBlock:^(ASCellNode *node, NSUInteger rowIndex, BOOL *stop) {
+      if (node.isNodeLoaded && node.view.superview) {
+        [node.view removeFromSuperview];
```

Comment:
This needs to be modified a little bit to handle layer-backed nodes.  I'm pretty sure an ASCellNode itself can be layer-backed.

We could provide a nicer API to hide this, but I think for now you need:

if (node.isLayerBacked)
   [node.layer removeFromSuperlayer];
else
   [node.view removeFromSuperview];